transform-plugins/src/main/java/co/cask/hydrator/plugin/Decoder.java

```diff
@@ -0,0 +1,223 @@
+
+  // Decoder handlers.
+  private final Base64 base64Decoder = new Base64();
+  private final Base32 base32Decoder = new Base32();
+  private final Hex hexDecoder = new Hex();
+
+  // Output Field name to type map
+  private final Map<String, Schema.Type> outSchemaMap = new HashMap<>();
+
+  // This is used only for tests, otherwise this is being injected by the ingestion framework.
+  public Decoder(Config config) {
+    this.config = config;
+  }
+
+  private void parseConfiguration(String config) throws IllegalArgumentException {
+    String[] mappings = config.split(",");
+    for (String mapping : mappings) {
+      String[] params = mapping.split(":");
+
+      // If format is not right, then we throw an exception.
+      if (params.length < 2) {
+        throw new IllegalArgumentException("Configuration '" + mapping + "' is incorrectly formed. " +
+                                             "Format should be <fieldname>:<decoder-type>");
+      }
+
+      String field = params[0];
+      String type = params[1].toUpperCase();
+      DecoderType eType = DecoderType.valueOf(type);
+
+      if (decodeMap.containsKey(field)) {
+        throw new IllegalArgumentException("Field " + field + " already has decoder set. Check the mapping.");
+      } else {
+        decodeMap.put(field, eType);
+      }
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    super.configurePipeline(pipelineConfigurer);
+    parseConfiguration(config.decode);
+    // Check if schema specified is a valid schema or no. 
+    try {
+      Schema.parseJson(config.schema);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.");
+    }
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    parseConfiguration(config.decode);
+    try {
+      outSchema = Schema.parseJson(config.schema);
+      List<Field> outFields = outSchema.getFields();
+      for (Field field : outFields) {
+        outSchemaMap.put(field.getName(), field.getSchema().getType());
+      }
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.");
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord in, Emitter<StructuredRecord> emitter) throws Exception {
+    StructuredRecord.Builder builder = StructuredRecord.builder(outSchema);
+
+    Schema inSchema = in.getSchema();
+    List<Field> inFields = inSchema.getFields();
+
+    // Iterate through input fields. Check if field name is present 
+    // in the fields that need to be decoded, if it's not then write
+    // to output as it is. 
+    for (Field field : inFields) {
+      String name = field.getName();
+
+      // Check if output schema also have the same field name. If it's not 
```

Comment:
this comment doesn't seem match the code. its doing `continue` instead of throwing exception.